aws-connect

```diff
@@ -208,6 +224,70 @@ elif [ "${action}" == "tunnel" ]; then
     --parameters "{\"portNumber\":[\"22\"],\"localPortNumber\":[\"${port}\"]}" \
     --region "${aws_region}" \
     --profile "${aws_profile}"
+elif [ "${action}" == "document" ]; then
+
+  if [ -z "${github_token_location}" ]
+  then
+    echo "No github token provided - scripts referenced in SSM document must be in a public repo if no token is provided"
+  else
+    arguments="githubTokenLocation='${github_token_location}',"
+  fi
+
+  if [ -z "${document_parameters}" ]
+  then
+    arguments="${arguments::${#arguments}-1}"
+  else
+    arguments="${arguments}parameters='${document_parameters}'"
+  fi
+
+  if [ -z "${cloudwatch_group}" ]
+  then
+    cloudwatch_group="aws-connect"
+  fi
+
+  echo "Running ${document_name} on instance ${instance_id}"
+
+  sh_command_id=$( aws ssm send-command \
+    --instance-ids ${instance_id} \
+    --document-name "${document_name}" \
+    --comment "Document run using aws-connect" \
+    --output text \
+    --parameters "${arguments}" \
+    --region ${aws_region} \
+    --profile ${aws_profile} \
+    --cloud-watch-output-config '{"CloudWatchLogGroupName":"'${cloudwatch_group}'","CloudWatchOutputEnabled":true}' \
+    --query "Command.CommandId")
+
+  status=$(aws ssm list-command-invocations \
+    --command-id "${sh_command_id}" \
+    --details \
+    --output text \
+    --no-paginate \
+    --query "CommandInvocations[0].Status")
+
+  echo "Command ID: $sh_command_id"
+
+  # wait for the command to finish running before getting the output
+  while [ "$status" == "InProgress" ]
```

Comment:
Ah I've been bitten by this before.  If the command never gets to InProgress, you'll never drop into the while loop.  For example, if it starts in `Pending` or `Delayed `, you will never loop.

Check out all the states
https://docs.aws.amazon.com/systems-manager/latest/userguide/monitor-commands.html

I also found that sometimes when SSM is busy, there is just no status!  So loop again if the status is not set

I found a methodology like this worked in another solution I did (this is python) for example

```
ssm_failed_statuses = ['Cancelled', 'TimedOut', 'Failed']
: 
: 
# Do we have a status?
                    if 'CommandInvocations' in command_status:
                        # Check if the command worked
                        if command_status_string == 'Success':
                            instances_command_status[instance] = 'SUCCESS'
                        elif command_status_string in ssm_failed_statuses:
                            status_code = command_status['ResponseCode']
                            instances_command_status[instance] = 'FAILED-' + str(status_code) 
                            failed_commands = True
```

Basically, keep looping until you get a success or a failed status